fio/lib/src/utils.ts

```diff
@@ -0,0 +1,281 @@
+    abi_hash:
+      "19a099047a11ee6c2a1d86b44b9100eba44ef85b60e33edfbdfcc58ff6ab7c56",
+    account_name: "fio.address",
+    code_hash:
+      "72c258fbcb0b328960bdbc4fc521d76cc53e5517bb5f5f2bf991d08030ddae6b",
+  };
+
+  const abiMap = new Map<string, FioTokenAbi>();
+  abiMap.set(account, fioTokenAbi);
+
+  const abiProvider: AbiProvider = {
+    getRawAbi: async () => {
+      const rawAbi = abiMap.get(account);
+      if (!rawAbi) throw new Error("Missing abi for " + account);
+      const abi = base64ToBinary(rawAbi.abi);
+      const binaryAbi = {
+        accountName: rawAbi.account_name,
+        abi: abi,
+      };
+      return binaryAbi;
+    },
+  };
+
+  return { fioTokenAbi, abiProvider };
+};
+
+/**
+ * Gets chain data.
+ * @returns The chain data.
+ */
+const getChainData = async (fioBaseUrl: string): Promise<ChainData> => {
+  const chainInfo = await fetch(`${fioBaseUrl}/chain/get_info`, {
```

Comment:
fetch doesn't fail even when server returns non-200. Let's use axios. It takes cares of these cases.

Reply:
done